### PR TITLE
create nav bar styling and routes

### LINKED ISSUE #11: Add a menu bar that dynamically displays other pages in the app
TBD

### PR DESCRIPTION
**Pull request by AnshaalHussain**
## Description

Created navbar that redirects to website url paths such as _map_, _home_, and _about us_

## Tasks

- created flask routes in ```__init__.py```
- added navbar in ```index.html``` template
- added navbar styling in ```main.css```
- added pages data list in ```app/data.json```

## Resources and screenshots (optional)

https://hackersandslackers.com/flask-routes/

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -11,7 +11,17 @@
...
+
+
+@app.route('/map')
+def map():
+    return "MAP SHOULD BE HERE"
+
+
+@app.route('/aboutme')
+def aboutme():
+    return "ABOUT ME"
```
Comment (dmosc) [inline app/__init__.py]: I think we might leave this off to the pr that actually implements them. It's a good idea to ship code that does a task. Maybe leave TODO comments?

```
// TODO: Implement /map route.
// TODO: Implement /aboutme route.
```

File: app/data.json [hunk 2]
```diff
@@ -62,5 +62,8 @@
         "Python"
       ]
     }
-  ]
+  ],
+  
+  "pages": [{"name": "Home", "url": "http://127.0.0.1:5000/"}, {"name": "Map", "url": "http://127.0.0.1:5000/map"}, {"name": "About Us", "url": "http://127.0.0.1:5000/aboutme"}]
```
Comment (dmosc) [inline app/data.json]: I first intended to use this file to "mirror" a database that fetches information related to our users. The "configuration" parameter you're suggesting would fit better in a file like: `app/config.json` (which is something I'm totally down for).

Else, I think hard-coding these values inside the HTML sounds like a straightforward solution as well.

File: app/static/styles/main.css [hunk 3]
```diff
@@ -31,6 +31,22 @@ body {
...
+
+}
+
+.navbar-content {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 30px 4%;
+}
+
```
Comment (dmosc) [inline app/static/styles/main.css:59]: All this is related to the previous comment on implementing a feature altogether.

File: app/static/styles/main.css [hunk 4]
```diff
@@ -41,6 +57,23 @@ body {
...
+
+.nav__links li a:hover {
+
+color: var(--yellow)
+}
+
+
+
```
Comment (dmosc) [inline app/static/styles/main.css]: On another note, I think we should stick to a uniform naming convention for CSS. All existing styles use `dash-names` :))

File: app/templates/index.html [hunk 5]
```diff
@@ -23,15 +23,24 @@
...
             <img width="200px" src="/static/images/mlh-logo.svg" />
           </div>
         </a>
+        <nav>
+          <ul class="nav__links">
+            {% for page in pages %}
+            <li><a href="{{ page.url }}">{{ page.name }}</a></li>
+            {% endfor %}
+          </ul>
+        </nav>
```
Comment (dmosc) [inline app/templates/index.html]: ^

**Review by dmosc — CHANGES_REQUESTED** [review]
Hi, thanks for the work!

Left some comments. Also, I simultaneously opened the [3d rendering pr](https://github.com/MLH-Fellowship/project-placeholder/pull/21) and some files are going to conflict on the pr that gets merged second. I think we should pick one to be merged first and the other will have to resolve conflicts.

Let me know what sounds best for you.

**Review by LauraAurora — APPROVED** [review]

File: app/data.json (hunk 2 above)
Comment (LauraAurora) [reply app/data.json]: I'm open to both solutions.

Comment (AnshaalHussain) [reply app/data.json]: I'm interested in the ```app/config.json``` idea! I need some more clarification on what that would look like however.

File: app/__init__.py (hunk 1 above)
Comment (AnshaalHussain) [reply app/__init__.py]: Makes sense, I'll replace them with TODO comments. I had made them to make sure routing functionality works with the nav bar links.

File: app/data.json (hunk 2 above)
Comment (AnshaalHussain) [reply app/data.json]: I've hardcoded the URLs values for now.

File: app/static/styles/main.css (hunk 3 above)
Comment (AnshaalHussain) [reply app/static/styles/main.css:59]: Sorry, I'm not sure what you mean by this comment.

Comment (dmosc) [reply app/static/styles/main.css:59]: Oh actually nvm, these are rules for the navbar; my bad.

File: app/static/styles/main.css [hunk 6]
```diff
@@ -41,6 +57,23 @@ body {
...
+}
+
+
+.nav-links li a:hover {
+
+color: var(--yellow)
+}
+
+
+
```
Comment (dmosc) [inline app/static/styles/main.css:86]: rm

**Review by dmosc — APPROVED** [review]
Nice; thank you so much for the effort. It looks awesome.

Just have to resolve the conflicts. If you think it might help, we could definitely set up a pair programming session and I can help you out.

**Pull request merged**